### ISSUE TITLE
 The mx-bluesky documentation is confusing, misleading and repetitive

### ISSUE THREAD

**Issue by rtuck99** [DiamondLightSource/mx-bluesky#578]
Some additional comments about the documentation - I think it's still confusing, partly the fault of python copier template. Perhaps in future we should just ignore most of their documentation templates because I think they are not really designed for our use case.

So, if you go to the documentation front page, to look for how to deploy mx-bluesky/hyperion you see this:

![image](https://github.com/user-attachments/assets/f1cfbaf2-9639-4f39-9da6-10a450e0efdb)

ok, so go to the "User Guide"
![image](https://github.com/user-attachments/assets/54858912-30a7-4ecd-b16e-7e779351354e)

under installation, this doesn't mention `dls_dev_env.sh`.

the real documentation is in mx-bluesky/build/html/developer/general/how-to/get-started.html

I think we should delete the installation documentation in the User Guide as generally users won't be installing mx-bluesky themselves either as a development install or a production deployment.

Similarly "Run in a container"

In fact, maybe we should just get rid of the User Guide section entirely as there is nothing in there at the moment for actual users.

Ok, so how about the Developer Guide:
![image](https://github.com/user-attachments/assets/36c6bca7-f70b-4e9e-ba85-12df99127b54)

Maybe we should change the title of General MX-Bluesky Developer Guide so we don't have two of them...

So lets try General MX-Bluesky Developer Guide (the first one):

![image](https://github.com/user-attachments/assets/9cd04f86-7e7c-4790-8e37-8524c2dafe61)

If I'm a developer, the install documentation must be in Developer Install? Wrong!

That's just the generic install documentation

Try Get started with mx-bluesky - at least this tells you to use `dls_dev_setup.sh`. But wait - it's really called  `dls_dev_env.sh`. 

...to be continued

**Comment by olliesilvester** [comment]
@rtuck99 I agree that the docs are gross, are you happy for me to pick this up?

**Comment by olliesilvester** [comment]
I agree with most of your points, but I think the hyperion developer guide should be a subcategory of the general mx-bluesky developer guide, rather than renaming one of them

**Comment by olliesilvester** [comment]
Going to split this up into the following:

- Remove user guide, and in the main page explain that we have no users right now so we only need a developer guide
- Put Hyperion developer guide as a subcategory of genera mx-bluesky developer guide
- Put the Hyperion overview (https://diamondlightsource.github.io/mx-bluesky/main/developer/hyperion/reference/readme.html) at the top level of the Hyperion part of the docs
- Update dev install instructions and remove duplication
- Re-add things from the old Hyperion wiki, like the code-map
- Then, why I originally started looking at this... Add a page for running zocalo system tests, how to use logging, how to update the panda IOC

**Comment by olliesilvester** [comment]
The difference between 'tutorials' and 'how-to-guides' is a bit confusing. According to https://docs.divio.com/documentation-system/, tutorials are 'learning based' and how-to guides are 'objective based'.

I interpret this to mean tutorials should teach you about concepts specific to this repo - like usage of helper classes we have made; while how-to guides should explain how to actually achieve things - like creating and deploying a release

More examples:

**Tutorials**
- Creating an ophyd-async device 
- Writing and running tests with pytest
- Using the parameter model
- Using `PlanReactiveCallbacks`

**How to guides**
- Getting started with development
- Creating and deploying new releases

**Comment by rtuck99** [comment]
When we move over the Wiki to the new documentation, we should remember to update all the links in the code comments to point to the new documentation.
